interfaces/mount/mountinfo.go

```diff
@@ -36,3 +42,64 @@ type InfoEntry struct {
 	MountSource  string
 	SuperOpts    string
 }
+
+// ParseInfoEntry parses a single line of /proc/$PID/mountinfo file.
+func ParseInfoEntry(s string) (InfoEntry, error) {
+	var e InfoEntry
+	var err error
+	fields := strings.Fields(s)
+	// The format is variable-length, but at least 10 fields are mandatory.
+	// The (7) below is a list of optional field which is terminated with (8).
+	// 36 35 98:0 /mnt1 /mnt2 rw,noatime master:1 - ext3 /dev/root rw,errors=continue
+	// (1)(2)(3)   (4)   (5)      (6)      (7)   (8) (9)   (10)         (11)
+	if len(fields) < 10 {
+		return e, fmt.Errorf("incorrect number of fields, expected at least 10 but found %d", len(fields))
+	}
+	// Parse MountID (decimal number).
+	e.MountID, err = strconv.Atoi(fields[0])
+	if err != nil {
+		return e, fmt.Errorf("cannot parse mount ID: %q", fields[0])
```

Comment:
The errors from strconv are terribly useless. We did this exercise in fstab parser. I honestly think that any human being can see if this is not a number.